### PR TITLE
Change time-related flags to durations

### PR DESCRIPTION
**Pull request by thockin**
Builds on #298 

Add '--period' to replace '--wait', which is now obsolete.
    
Add '--sync-timeout' to replace '--timeout', which is now obsolete.
    
Both of these new flags take a Go-style time string, rather than a bare
number. For example "1s" for 1 second or "1m" for one minute.
    
The old flags have been kept and will take precedence if specified.

xref #297

### CODE REVIEW

File: cmd/git-sync/main.go [hunk 1]
```diff
@@ -1145,11 +1160,13 @@ OPTIONS
             An optional command to be executed after syncing a new hash of the
             remote repository.  This command does not take any arguments and
             executes with the synced repo as its working directory.  The
-            execution is subject to the overall --timeout flag and will extend
-            the period between syncs attempts.
+            execution is subject to the overall --sync-timeout flag and will
+            extend the effective period between syncs attempts.
```
Comment (stp-ip) [inline cmd/git-sync/main.go:1164]:
```suggestion
            extend the effective period between sync attempts.
```

**Review by stp-ip — CHANGES_REQUESTED** [review]
small nit

/approve

File: cmd/git-sync/main.go (hunk 1 above)
Comment (thockin) [reply cmd/git-sync/main.go:1164]: ACK

**Comment by thockin** [comment]
Fixed, rebased and repushed.  PTAL

**Pull request merged**